src/multi/bmp180.cpp

```diff
@@ -0,0 +1,305 @@
+  m_calibration_data.ac4 =
+    (calibration_data_buffer[6] << 8) | calibration_data_buffer[7];
+  m_calibration_data.ac5 =
+    (calibration_data_buffer[8] << 8) | calibration_data_buffer[9];
+  m_calibration_data.ac6 =
+    (calibration_data_buffer[10] << 8) | calibration_data_buffer[11];
+  m_calibration_data.b1 =
+    (calibration_data_buffer[12] << 8) | calibration_data_buffer[13];
+  m_calibration_data.b2 =
+    (calibration_data_buffer[14] << 8) | calibration_data_buffer[15];
+  m_calibration_data.mb =
+    (calibration_data_buffer[16] << 8) | calibration_data_buffer[17];
+  m_calibration_data.mc =
+    (calibration_data_buffer[18] << 8) | calibration_data_buffer[19];
+  m_calibration_data.md =
+    (calibration_data_buffer[20] << 8) | calibration_data_buffer[21];
+}
+
+hal::celsius bmp180::temperature()
+{
+  // run helper function which computes the temperature and b5
+  auto temperature_data = compute_temperature_and_b5(m_i2c,
+                                                     m_address,
+                                                     m_calibration_data.ac5,
+                                                     m_calibration_data.ac6,
+                                                     m_calibration_data.mc,
+                                                     m_calibration_data.md);
+
+  // return just the temperature data
+  return temperature_data.temperature;
+}
+
+bmp180::pressure_results bmp180::pressure(int sample_amount)
+{
+  // clamp sample amount to determined maximum
+  sample_amount = std::clamp(sample_amount, 1, m_maximum_samples);
+
+  computation_variables variables;
+  // run helper function which computes the temperature and b5
+  auto temperature_data = compute_temperature_and_b5(m_i2c,
+                                                     m_address,
+                                                     m_calibration_data.ac5,
+                                                     m_calibration_data.ac6,
+                                                     m_calibration_data.mc,
+                                                     m_calibration_data.md);
+
+  // tell sensor to start conversion for pressure
+  hal::byte control_setting =
+    (static_cast<hal::byte>(m_oversampling_setting) << 6) | 0x34;
+  std::array<hal::byte, 2> sensor_config_buffer = {
+    control_measurement_register()[0], control_setting
+  };
+
+  hal::byte oversampling_setting =
+    static_cast<hal::byte>(m_oversampling_setting);
+  std::int32_t summed_uncompensated_pressures = 0,
+               average_uncompensated_pressure = 0;
+
+  // loop to wait for conversion to complete
+  int conversion_finished;
+  for (int i = 0; i < sample_amount; i++) {
+    conversion_finished = 0;
+    hal::write(*m_i2c, m_address, sensor_config_buffer, hal::never_timeout());
+    while (conversion_finished == 0) {
+      conversion_finished = conversion_status(m_i2c, m_address);
+    }
```

Comment:
```suggestion
  for (int i = 0; i < sample_amount; i++) {
    hal::write(*m_i2c, m_address, sensor_config_buffer, hal::never_timeout());
    int conversion_finished = 0;
    while (conversion_finished == 0) {
      conversion_finished = conversion_status(m_i2c, m_address);
    }
```

Keep variables close the scope where they are used. And refrain from uninitialized variables.

Reply:
It may even be better to simply put this into its own function since you tend to need it a few times in the code. The function will spin until the conversion is done.